azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -792,6 +824,13 @@ func resourceArmKubernetesClusterUpdate(d *schema.ResourceData, meta interface{}
 		existing.ManagedClusterProperties.WindowsProfile = windowsProfile
 	}
 
+	if d.HasChange("managed_cluster_identity") {
+		updateCluster = true
+		managedClusterIdentityRaw := d.Get("managed_cluster_identity").([]interface{})
+		managedClusterIdentity := expandKubernetesClusterManagedClusterIdentity(managedClusterIdentityRaw)
+		existing.Identity = managedClusterIdentity
```

Comment:
minor but this could be
```
		existing.Identity = expandKubernetesClusterManagedClusterIdentity(d.Get("managed_cluster_identity")) 
```

if the expand sig is changed